modules/core/src/lib/layer.js

```diff
@@ -803,22 +803,7 @@ ${flags.viewportChanged ? 'viewport' : ''}\
     // TODO - is attribute manager needed? - Model should be enough.
     const attributeManager = this.getAttributeManager();
     const attributeManagerNeedsRedraw = attributeManager && attributeManager.getNeedsRedraw(opts);
-    const modelNeedsRedraw = this._modelNeedsRedraw(opts);
-    redraw = redraw || attributeManagerNeedsRedraw || modelNeedsRedraw;
-
-    return redraw;
-  }
-
-  _modelNeedsRedraw(opts) {
-    let redraw = false;
-
-    for (const model of this.getModels()) {
```

Comment:
To keep supporting luma.gl property animation, we would simply iterate over the models and check `model.animated`.

Reply:
This is controlled by deck prop `_animated`. I don't think any model should set its `animated` flag independently from that.

Reply:
Since this feature is considered experimental in deck.gl we can certainly choose to limit support for it by making this change, but here are my thoughts:

>  I don't think any model should set its animated flag independently from that.

The `Model.animated` "flag" is simply indicating whether any uniform supplied to `Model.setUniforms()` is a function, meaning that multiple calls to `Model.draw()` may draw different things.

Setting aside the question whether the luma.gl uniform animation feature is valuable or worthwhile:

- A model that has a function valued uniform will (typically) draw different things every time it is called.
- If it is not redrawn every frame, but only when redraw is triggered for other reasons would lead to a random animation state being left on the screen.
- Requiring the app to also set the top-level `animated` prop makes it more cumbersome to use this feature, in some cases it could require the app to track this state on its own.